src/main/java/org/spongepowered/api/item/inventory/Inventory.java

```diff
@@ -0,0 +1,263 @@
+    
+    /**
+     * Query this inventory for inventories containing any of the supplied item
+     * types. This query operates directly on {@link Slot} leaf nodes in the
+     * inventory and will always return a collection containing only {@link
+     * Slot} instances. Logical <code>OR</code> is applied between operands.
+     */
+    <T extends Inventory> T query(ItemTypes... types);
+    
+    /**
+     * Query this inventory for inventories containing any stacks which match
+     * the supplied stack operands. This query operates directly on {@link Slot}
+     * leaf nodes in the inventory and will always return a collection
+     * containing only {@link Slot} instances. Logical <code>OR</code> is
+     * applied between operands.
+     */
+    <T extends Inventory> T query(ItemStack... types);
+    
+    /**
+     * Query this inventory for inventories which match any of the supplied
+     * properties. The <code>equals</code> method of each property is called on
+     * each child inventory which has the supplied property. Logical
+     * <code>OR</code> is applied between operands. This method is effectively
+     * the same as calling {@link #query} with an {@link
+     * InventoryProperty.Operator} of {@link InventoryProperty.Operator#EQUAL}.
+     */
+    <T extends Inventory> T query(InventoryProperty<?, ?>... props);
+    
+    /**
+     * Query this inventory for inventories which match any of the supplied
+     * properties with the specified operator. The relevant method of each
+     * property is called on each child inventory which has the supplied
+     * property. Logical <code>OR</code> is applied between operands.
+     */
+    <T extends Inventory> T query(InventoryProperty.Operator op, InventoryProperty<?, ?>... props);
+    
+    /**
+     * Query this inventory for inventories matching any of the supplied titles.
+     * Logical <code>OR</code> is applied between operands.
+     */
+    <T extends Inventory> T query(Translatable... names);
+    
+    /**
+     * Query this inventory for inventories matching any of the supplied titles.
+     * Logical <code>OR</code> is applied between operands.
+     */
+    <T extends Inventory> T query(String... names);
+    
+    /**
+     * <p>Query this inventory by dynamically inspecting each operand. Each
+     * operand in turn is checked for a match against the other query methods,
+     * and if a matching method is found the query is performed using the
+     * operand. This is repeated until all operands are consumed and allows a
+     * union of multiple query types to be aggregated into a single view.</p>
+     * 
+     * <p>For operands with no matching type, the behaviour is determined by the
+     * individual inventory. A naive match may be obtained by calling .equals()
+     * against the child inventory passing the unknown operand as an argument.
+     * </p>
+     */
+    <T extends Inventory> T query(Object... args);
```

Comment:
No, because `Predicate` indicates that the logic is supplied by the user. The comparison logic is implicit in the `InventoryProperty` passed in and the logic is deterministically implemented by the inventory itself.